File: java/leetcode_grind/Day665.java
package leetcode_grind;

public class Day665 {
    // https://leetcode.com/problems/count-the-number-of-consistent-strings/description/?envType=daily-question&envId=2024-09-12
    static class Solution1 {
        public int countConsistentStrings(String allowed, String[] words) {
            int allowedBits = 0;
            for (int i = 0; i < allowed.length(); i++) {
                allowedBits |= 1 << (allowed.charAt(i) - 'a');
            }

            int consistentCount = 0;
            for (String word : words) {
                boolean isConsistent = true;

                for (int i = 0; i < word.length(); i++) {
                    int bit = (allowedBits >> (word.charAt(i) - 'a')) & 1;
                    if (bit == 0) {
                        isConsistent = false;
                    }
                }

                if (isConsistent) {
                    consistentCount++;
                }
            }
            return consistentCount;
        }
    }
}

File: src/leetcode_grind/day_665.rs
// https://leetcode.com/problems/count-the-number-of-consistent-strings/description/?envType=daily-question&envId=2024-09-12
pub fn count_consistent_strings(allowed: String, words: Vec<String>) -> i32 {
    use std::collections::HashSet;
    let allowed = allowed.chars().collect::<HashSet<_>>();
    let mut ans = 0;
    'lb: for w in words {
        let w = w.chars().collect::<HashSet<_>>();
        for c in &w {
            if !allowed.contains(c) {
                continue 'lb;
            }
        }
        ans += 1;
    }
    ans
}
